Sources/Orbit/Components/Collapse.swift

```diff
@@ -0,0 +1,129 @@
+import SwiftUI
+
+struct CollapseButtonStyle: SwiftUI.ButtonStyle {
+
+    func makeBody(configuration: Configuration) -> some View {
+        configuration.label
+            .contentShape(Rectangle())
+    }
+}
+
+/// Hides long or complex information.
+///
+/// - Note: [Orbit definition](https://orbit.kiwi/components/collapse/)
+public struct Collapse<Header: View, Content: View>: View {
+
+    let header: Header
+    let content: Content
+
+    @State var isExpanded = false
+
+    public var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            SwiftUI.Button {
+                withAnimation(.easeInOut(duration: 0.2)) {
+                    isExpanded.toggle()
+                }
+            } label: {
+                HStack(spacing: 0) {
+                    header
+                        .accessibility(.collapseHeader)
+                    Spacer(minLength: .xSmall)
+                    Icon(.chevronDown)
+                        .rotationEffect(isExpanded ? .degrees(180) : .zero)
+                }
+                .padding(.vertical, .small)
+            }
+            .buttonStyle(CollapseButtonStyle())
+
+            if isExpanded {
+                content
+                    .padding(.vertical, .small)
+            }
+        }
+        .overlay(Separator(), alignment: .bottom)
+    }
+}
+
+public extension Collapse {
+
+    /// Creates Orbit Collapse component.
+    init(isExpanded: Bool = false, @ViewBuilder content: () -> Content, @ViewBuilder headerContent: () -> Header) {
+        self.header = headerContent()
+        _isExpanded = .init(initialValue: isExpanded)
```

Comment:
If there is a chance that a call site would like to be informed about the state change, we would probably switch to a @Binding in future. We currently mix multiple approaches in our API already - a future follow-up might clean-up those that are less easy to work with. 

I expect that in future, we might replace some of our @State properties for @Bindings (or properties + callbacks) as we often later find out that a call site needs to be able to bind that state with its own mutable single source of truth.

Reply:
Whether view is expanded or not seems to fit the definition of `@State`... I wouldn't want to force users to pass in a `@Binding` in all circumstances - is there a way to get around that? If not, I'd probably go with a callback.

Reply:
Hmm... `DisclosureGroup` does have an init that takes a `Binding`... and also one that doesn't... I wonder how that works.

Reply:
I remember we did (or tried?) this already in some component - in the same way, there is an init both for a "normal" property and for binding. I believe two different internal properties are used under the hood at the worst case.

Another components may benefit from this double init (especially `Tag`). I will create a follow-up, but feel free to introduce the swiftui-like api in this story to see how it would work.

Reply:
Found [this article](https://www.fivestars.blog/articles/optional-binding/) which explores the same topic (and with a `DisclosureGroup` too, of course 😄) - used the ideas from it here.